ReminderApp-OBJC/ReminderApp-OBJC/Classes/Controller/CDReminderTableViewController.m

```diff
@@ -129,62 +101,231 @@ - (void)controller:(NSFetchedResultsController *)controller didChangeObject:(id)
 	}
 	
 }
 
-- (void) controllerDidChangeContent:(NSFetchedResultsController *)controller {
+- (void)controllerDidChangeContent:(NSFetchedResultsController *)controller {
 	
 	[self.tableView endUpdates];
 	
 }
 
-#pragma mark - DataSource and Delegate
+#pragma mark - Helper methods
+
+- (CGFloat)textViewHeightForRowAtIndexPath:(NSIndexPath *)indexPath {
 
-- (NSInteger)numberOfSectionsInTableView:(UITableView *) tableView {
+	UITextView *calculationView = ((CDAddReminderCell *)[self.tableView cellForRowAtIndexPath:indexPath]).textView;
+	CGFloat textViewWidth = calculationView.frame.size.width;
 	
-	return self.fetchedResultsController.sections.count;
+	if (!calculationView) {
+		return 50;
+	}
 
+	CGSize size = [calculationView sizeThatFits:CGSizeMake(textViewWidth, FLT_MAX)];
+	return size.height;
+	
 }
 
-- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+- (void)scrollToCursorForTextView:(UITextView *)textView {
+	
+	CGRect cursorRect = [textView caretRectForPosition:textView.selectedTextRange.start];
+	cursorRect = [self.tableView convertRect:cursorRect fromView:textView];
+	
+	if (![self rectVisible:cursorRect]) {
+		cursorRect.size.height += 8;
+		[self.tableView scrollRectToVisible:cursorRect animated:YES];
+	}
+	
+}
+
+#pragma mark - Keyboard specific methods
+
+- (BOOL)rectVisible:(CGRect)rect {
+	
+	CGRect visibleRect;
+	visibleRect.origin = self.tableView.contentOffset;
+	visibleRect.origin.y += self.tableView.contentInset.top;
+	visibleRect.size = self.tableView.bounds.size;
+	visibleRect.size.height -= self.tableView.contentInset.top + self.tableView.contentInset.bottom;
+	
+	return CGRectContainsRect(visibleRect, rect);
+	
+}
+
+- (void)keyboardWillShow:(NSNotification *)aNotification {
+	
+	NSDictionary* info = [aNotification userInfo];
+	CGSize kbSize = [[info objectForKey:UIKeyboardFrameBeginUserInfoKey] CGRectValue].size;
+	
+	UIEdgeInsets contentInsets = UIEdgeInsetsMake(self.tableView.contentInset.top, 0.0, kbSize.height, 0.0);
+	
+	self.tableView.contentInset = contentInsets;
+	self.tableView.scrollIndicatorInsets = contentInsets;
+	
+}
+
+- (void)keyboardWillHide:(NSNotification*)aNotification {
+	
+	[UIView beginAnimations:nil context:nil];
+	[UIView setAnimationDuration:0.35];
+	
+	UIEdgeInsets contentInsets = UIEdgeInsetsMake(self.tableView.contentInset.top, 0.0, 0.0, 0.0);
+	
+	self.tableView.contentInset = contentInsets;
+	self.tableView.scrollIndicatorInsets = contentInsets;
+	
+	[UIView commitAnimations];
+	
+}
+
+#pragma mark - UITextViewDelegate
 
-	return self.fetchedResultsController.sections[section].numberOfObjects;
+- (void)textViewDidBeginEditing:(UITextView *)textView {
+	
+	NSLog(@"Begin editing \n");
+	[self scrollToCursorForTextView:textView];
+	
+}
+
+- (void)textViewDidEndEditing:(UITextView *)textView {
+	
+	NSLog(@"End edititing \n");
+	
+}
+
+- (void)textViewDidChange:(UITextView *)textView {
+	
+	NSLog(@"DidChange \n");
+	
+	[self.tableView beginUpdates];
```

Comment:
Remove this. After you change something in core data "beginUpdates" and "endUpdates" will be performed by the fetchResultController delegate methods.